lib/sdf-server/src/service/v2/admin/migrate_connections.rs

```diff
@@ -323,38 +327,80 @@ async fn add_prop_connection(
     Ok(true)
 }
 
-/// Remove the existing socket connection (unless it was inferred, in which case there isn't one)
-async fn remove_socket_connection(
+// Look at the target socket, and follow it to any props, and replace their values with
+// their current static value if they haven't already been set (which will happen if it was
+// unmigrateable).
+// (If we already migrated something, it will have its value set!)
+async fn freeze_connection_values(
     ctx: &DalContext,
-    explicit_connection_id: Option<AttributePrototypeArgumentId>,
     socket_connection: &Option<SocketConnection>,
 ) -> AdminAPIResult<bool> {
-    // We don't remove inferred connections
-    let Some(explicit_connection_id) = explicit_connection_id else {
-        return Ok(false);
-    };
-
-    // Remove the connection
-    AttributePrototypeArgument::remove(ctx, explicit_connection_id).await?;
+    let mut did_something = false;
 
-    // Send the WsEvent
     if let &Some(SocketConnection {
-        from: (from_component_id, from_socket_id),
         to: (to_component_id, to_socket_id),
+        ..
     }) = socket_connection
     {
+        for apa_id in
+            AttributePrototypeArgument::list_references_to_value_source(ctx, to_socket_id).await?
+        {
+            let prototype_id = AttributePrototypeArgument::prototype_id(ctx, apa_id).await?;
+            if let Some(prop_id) = AttributePrototype::prop_id(ctx, prototype_id).await? {
+                let av_id =
+                    Component::attribute_value_for_prop_id(ctx, to_component_id, prop_id).await?;
+                // Only set its value if it isn't already set!
+                if AttributeValue::component_prototype_id(ctx, av_id)
+                    .await?
+                    .is_none()
+                {
+                    // Pull the current value from the AV, and set it explicitly to that value
+                    // so it remains the same, but doesn't rely on the subscription anymore
```

Comment:
do you mean connection here, not subscription?

Reply:
I do!